test/distributed/test_c10d.py

```diff
@@ -3075,6 +3075,35 @@ def test_ddp_comm_hook_future_passing_gpu_nccl(self):
         """
         self._test_ddp_comm_hook_future_passing_gpu(c10d.ProcessGroupNCCL)
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_ddp_comm_hook_allreduce_hook_nccl(self):
+        """
+        This unit test verifies whether a DDP communication hook that just
+        calls allreduce gives the same result result with the case of no hook
+        registered.
+        """
+        store = c10d.FileStore(self.file_name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+
+        # Test on GPU
+        device_id = gpus_for_rank(self.world_size)[self.rank][0]
+        gpu_model = DistributedDataParallel(
+            TestDdpCommHook().to(device_id),
+            device_ids=[device_id],
+            process_group=process_group
+        )
+
+        def allreduce_hook(state: object, bucket: dist.GradBucket) -> torch.futures.Future:
+            return process_group.allreduce(bucket.get_tensors()).get_future()
```

Comment:
nit: Maybe divide the result by some constant after the allreduce is done to ensure its different from the default result. This just ensures the hook is being called correctly and not the default implementation inadvertently.

Reply:
I've included a then callback within allreduce_hook.